tests/zfs-tests/tests/functional/cli_root/zfs_mount/zfs_mount_test_race.sh

```diff
@@ -0,0 +1,93 @@
+#!/bin/ksh
+
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source.  A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright (c) 2019 by Tomohiro Kusumi. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+
+#
+# DESCRIPTION:
+# Verify parallel mount ordering is consistent.
+# See github.com/zfsonlinux/zfs/issues/8833 for details.
+#
+# STRATEGY:
+# 1. Create pools and filesystems.
+# 2. Set same mount point for >1 datasets.
+# 3. Unmount all datasets.
+# 4. Mount all datasets.
+# 5. Unmount all datasets.
+# 6. Mount all datasets.
+#
+
+verify_runnable "both"
+
+TMPDIR=${TMPDIR:-$TEST_BASE_DIR}
+MNTPT=$TMPDIR/zfs_mount_test_race_mntpt
+
+DISK1="$TMPDIR/zfs_mount_test_race_disk1"
+TESTPOOL1=zfs_mount_test_race_tp1
+TESTFS1=zfs_mount_test_race_tf1
```

Comment:
Do we need to redefine `TESTPOOL{1,2}` and `TESTFS{1,2}`, they're already defined in `default.cfg`.

Reply:
I used unique names since it `rm -rf /$TESTPOOL{1,2}`, and also to be defensive against other tests in `zfs_mount` category.

Reply:
I see.  So there shouldn't be hidden dependencies between tests so using the original names should be fine, but if you'd prefer the custom names just in case that's fine with me.

Reply:
I got why I used unique names for `TESTPOOL{1,2}`. If the test runs with default names, it will not want to `rm -rf /$TESTPOOL{1,2}` on cleanup to be defensive with other tests in `zfs_mount` category. Other tests could have created something to use throughout several tests.

But then this test won't succeed if `/testpool{1,2}` already has a dirent(s) in it. e.g. Running this test 2 times in a row at local fails on the second attempt with below for having a directory under `/testpool2` already.

So yes, it's safer to just go with unique names.

```
ERROR: zpool create -f testpool2 /var/tmp/zfs_mount_test_race_disk2 exited 1
mountpoint '/testpool2' exists and is not empty use '-m' option to provide a different default
```

```
# tree /testpool2/
/testpool2/
`-- testfs2

1 directory, 0 files
```